cmd/gardenlet/app/app.go

```diff
@@ -388,6 +389,12 @@ func (g *garden) Start(ctx context.Context) error {
 		return err
 	}
 
+	// TODO(timuthy, rfranzke): Drop this code when the FullNetworkPoliciesInRuntimeCluster feature gate gets promoted to GA.
+	log.Info("Reconciling network policies in the seed cluster")
+	if err := g.reconcileNetworkPoliciesInSeed(ctx, g.mgr.GetClient()); err != nil {
+		return err
+	}
```

Comment:
As far as I understand it, this creates `allow-all` or removes all network policies in the existing extension namespaces. However, this happens only on start-up of `gardenlet`. Extensions are dynamically managed, i.e. they can also be deployed at a later point in time when they are needed.
When testing the provider specific configurations I ran into this multiple times. It can be resolved by restarting `gardenlet`, but this is not that great in a productive environment. (A productive environment might have less dynamic extensions, though.)

Was this intentional?

Reply:
You are absolutely right. It is conceptually wrong to put this code in the start-up sequence for the reasons you mentioned. I have now moved it into the `ControllerInstallation` controller which is responsible for managing the extension namespaces and deployments. When the feature gate gets enabled/disabled, gardenlet is restarted and reconciles all such namespaces, and when a new extension gets created, then the same code is executed. This is the right place. That's a very nice catch, thanks for finding this! 🙇🏻